### DISCUSSION TITLE
Error: maximum tree depth (10) exceeded calculating gcc@5 dependency tree!

### DISCUSSION THREAD

**Discussion by jacktose** [Homebrew/discussions#3569] [Linux] [answered]
After the [change to glibc-bootstrap][glibc], most brew commands fail for me (CentOS 7, non-default path) with this error:
```
Error: maximum tree depth (10) exceeded calculating gcc@5 dependency tree!
```
`gcc@5` was not installed and hadn't needed to be for a while.
I tried a fresh install and have the same problem. [Full info is in a gist][gist], but here are some examples from immediately after the fresh install:
```
ad-user-ads-11 ~ ★ brew update
Already up-to-date.
ad-user-ads-11 ~ ★ brew list
ad-user-ads-11 ~ ★ brew search vim
Error: maximum tree depth (10) exceeded calculating gcc@5 dependency tree!
ad-user-ads-11 ~ ★ brew info gcc
Error: maximum tree depth (10) exceeded calculating gcc@5 dependency tree!
ad-user-ads-11 ~ ★ brew install gcc
Error: maximum tree depth (10) exceeded calculating gcc@5 dependency tree!
```

I've found the [source] for this error, but I don't know what actually causes it, and I find zero (0) search results for it. Help?

[glibc]: https://github.com/Homebrew/homebrew-core/pull/106837
[gist]: https://gist.github.com/jacktose/4d9c7799d721797c3f72c9765920db24
[source]: https://github.com/Homebrew/brew/blob/61db6207d8d9cd652b182fc7f183ffebab4ac49c/Library/Homebrew/extend/os/linux/dependency_collector.rb#L62

**Comment by carlocab** [comment]
Can you try to change `maximum_tree_depth` and work out the lowest you can make it without errors?

Sorry for the breakage!

**Reply by jacktose** [reply]
11 works!

And no worries. I'm excited about glibc-bootstrap and hopefully less trouble maintaining the build dependency chain.

**Comment by jacktose** [comment, marked as answer]
Ah! This was reported in https://github.com/Homebrew/brew/issues/13760 and fixed in https://github.com/Homebrew/brew/pull/13761. From the code comment, I suppose 20 is as good as 11 for a guard against infinite loops.
